lib/utils/node_id.dart

```diff
@@ -5,15 +5,29 @@ String parseNodeId(String nodeID) {
   if (nodeID == null) {
     return null;
   }
-  if (nodeID.length == NODE_ID_LENGTH) {
+  nodeID = nodeID.trim();
+
+  if (nodeID.length == NODE_ID_LENGTH && _isParsable(nodeID)) {
     return nodeID;
   }
 
   if (nodeID.length > NODE_ID_LENGTH &&
       nodeID.substring(NODE_ID_LENGTH, NODE_ID_LENGTH + 1) ==
           NODE_URI_SEPARATOR) {
-    return nodeID.substring(0, 66);
+    nodeID = nodeID.substring(0, NODE_ID_LENGTH);
+    if (_isParsable(nodeID)) {
+      return nodeID;
+    }
   }
-
   return null;
 }
+
+bool _isParsable(String nodeID) {
```

Comment:
Perhaps better to use the` hex.Decode `?

Reply:
Good point, changed it to use hex.decode in the latest rebase.